### PR TITLE
fix: Fix user update in client method and manager's mutation method

### LINKED ISSUE #717: Cannot update admin
### What Operating System(s) are you seeing this problem on?

macOS (Apple Silicon)

### Backend.AI version

dcf198c44b07aa348cf719772000c51e0ba5f6ac

### Describe the bug

If the user is admin or superadmin, it cannot update any option because of the below error.
image

### To Reproduce

1. Add an admin or superadmin user.
2. Update info of that user using `./backend.ai admin user update [OPTIONS] EMAIL`.
3. See an error.

### Expected Behavior

Update option without error (admin or superadmin previleged).

### Anything else?

I think [these codes](https://github.com/lablup/backend.ai/blob/main/src/ai/backend/manager/models/user.py#L696-L697) are the problem.

### PR DESCRIPTION
**Pull request by fregataa**
resolves #717 
change default parameters of user update method to `None`
fix and refactor user mutate method

### CODE REVIEW

**Review by Sujin-Kim1 — APPROVED** [review]
LGTM.

**Review by Sujin-Kim1 — APPROVED** [review]

File: src/ai/backend/manager/models/user.py [hunk 1]
```diff
@@ -655,13 +655,14 @@ async def _post_func(conn: SAConnection, result: Result) -> Row:
...
                     # User's becomes admin. Set the keypair as active admin.
+                    # TODO: Should we update the role of all users related to keypair?
                     kp = result.first()
                     kp_data = dict()
                     if not kp.is_admin:
                         kp_data["is_admin"] = True
                     if not kp.is_active:
                         kp_data["is_active"] = True
-                    if len(kp_data.keys()) > 0:
+                    if kp_data:
```
Comment (kyujin-cho) [inline src/ai/backend/manager/models/user.py:665]: Treating non-bool values as bool decreases code readability. Unless you do not have specific reason to do so, let's revert changes on this line.
```suggestion
                    if len(kp_data.keys()) > 0:
```

File: src/ai/backend/manager/models/user.py [hunk 2]
```diff
@@ -670,36 +671,34 @@ async def _post_func(conn: SAConnection, result: Result) -> Row:
...
                             kp_updates.append(kp_data)
-                        cnt += 1
-                    await conn.execute(
-                        sa.update(keypairs)
-                        .values(
-                            {
-                                "is_admin": bindparam("is_admin"),
-                                "is_active": bindparam("is_active"),
-                            }
+                    if kp_updates:
```
Comment (kyujin-cho) [inline src/ai/backend/manager/models/user.py:690]: Oops! Same here. 
```suggestion
                    if len(kp_updates) > 0:
```

**Review by kyujin-cho — CHANGES_REQUESTED** [review]

File: src/ai/backend/manager/models/user.py (hunk 1 above)
Comment (fregataa) [reply src/ai/backend/manager/models/user.py:665]: This one is Python's truthness
https://docs.python.org/3/library/stdtypes.html#truth-value-testing
https://github.com/lablup/backend.ai-manager/pull/563/commits/69d8467d6d064e704199ab51c01a55e150e9ff41

File: src/ai/backend/manager/models/user.py (hunk 2 above)
Comment (fregataa) [reply src/ai/backend/manager/models/user.py:690]: I already updated this one, but how about using Python truthness?

File: src/ai/backend/manager/models/user.py (hunk 1 above)
Comment (kyujin-cho) [reply src/ai/backend/manager/models/user.py:665]: Oh, I couldn’t catch the latter one. Looks like our current convention is to use Python’s truthness… So I guess you can just ignore this review and revert already changed one. Sorry for inconvenience!

Comment (fregataa) [reply src/ai/backend/manager/models/user.py:665]: thank you. I reverted the commit and you can re-review this.

**Review by kyujin-cho — APPROVED** [review]
LGTM. We still need to decide what to do when user reactivates inactive keypair which was created with previous user role, but it's out of this PR's scope.

**Pull request merged**